modules/oauth/src/main/java/org/mule/module/oauth2/internal/authorizationcode/DefaultAuthorizationCodeGrantType.java

```diff
@@ -256,8 +259,18 @@ public void start() throws MuleException
         }
         if (tokenRequestHandler != null)
         {
-            tokenRequestHandler.setOauthConfig(this);
+            tokenRequestHandler.setOauthConfig(this, proxyConfig);
             tokenRequestHandler.init();
         }
     }
+
+    public ProxyConfig getProxyConfig()
```

Comment:
Shouldn't this be higher up in the hierarchy considering both types use it? Seems like it should be part of the config element.

Reply:
Well, I can't see a configuration element where putting the reference to the proxy configuration. For example the tlsContext is referenced at this same level. 
Regarding the object model, I could put the proxyConfig fields and its respective getters and setters in the abstract class, but it looks like an interface. (There are others shared attributes like the tlsContext that should also be there).
At last, the tlsContext can be configured as a nested element in the gran type configuration, I am not sure if the same approach applies to the proxy configuration (I mean in the minor/major release and not here where we are only defining spring properties).

Reply:
I don't understand why it cannot go into `AbstractGrantType` instead of being repeated in both types.